src/shares.rs

```diff
@@ -176,10 +175,14 @@ pub struct SetInProgress {
     id_set: Vec<u32>,
     content_length: usize,
     content_set: Vec<Vec<u8>>,
+    nonce: String,
 }
 
 #[derive(Debug)]
-pub struct SetCombined(Vec<u8>);
```

Comment:
Adding a few doc comment lines for the purpose of this and its members would be appreciated.